apidef/api_definitions.go

```diff
@@ -596,7 +601,7 @@ type APIDefinition struct {
 	ExpireAnalyticsAfter       int64                  `mapstructure:"expire_analytics_after" bson:"expire_analytics_after" json:"expire_analytics_after"` // must have an expireAt TTL index set (http://docs.mongodb.org/manual/tutorial/expire-data/)
 	ResponseProcessors         []ResponseProcessor    `bson:"response_processors" json:"response_processors"`
 	CORS                       CORSConfig             `bson:"CORS" json:"CORS"`
-	Domain                     string                 `bson:"domain" json:"domain"`
+	Domain                     Domain                 `bson:"domain" json:"domain"`
```

Comment:
I think this breaks backward compatibility. can we keep `domain` as it is, and add `disable_custom_domain bool` as another field, so that custom domain will be enabled by default with default value `false` for `disable_custom_domain`